sdk/metric/pipeline.go

```diff
@@ -134,3 +140,189 @@ func (p *pipeline) produce(ctx context.Context) (metricdata.ResourceMetrics, err
 		ScopeMetrics: sm,
 	}, nil
 }
+
+// pipelineRegistry manages creating pipelines, and aggregators.  Meters retrieve
+// new Aggregators from a pipelineRegistry.
+type pipelineRegistry[N int64 | float64] struct {
+	views     map[Reader][]view.View
+	pipelines map[Reader]*pipeline
+}
+
+func newPipelineRegistries(views map[Reader][]view.View) (*pipelineRegistry[int64], *pipelineRegistry[float64]) {
+	pipelines := map[Reader]*pipeline{}
+	for rdr := range views {
+		pipe := &pipeline{}
+		rdr.register(pipe)
+		pipelines[rdr] = pipe
+	}
+	return &pipelineRegistry[int64]{
+			views:     views,
+			pipelines: pipelines,
+		}, &pipelineRegistry[float64]{
+			views:     views,
+			pipelines: pipelines,
+		}
+}
+
+// createAggregators will create all backing aggregators for an instrument.
+// It will return an error if an instrument is registered more than once.
+// Note: There may be returned aggregators with an error.
+func (reg *pipelineRegistry[N]) createAggregators(inst view.Instrument, instUnit unit.Unit) ([]internal.Aggregator[N], error) {
+	var aggs []internal.Aggregator[N]
+
+	errs := &multierror{}
+	for rdr, views := range reg.views {
+		pipe := reg.pipelines[rdr]
+		rdrAggs, err := createAggregators[N](rdr, views, inst)
+		if err != nil {
+			errs.append(err)
+		}
+		for inst, agg := range rdrAggs {
+			err := pipe.addAggregator(inst.scope, inst.name, inst.description, instUnit, agg)
+			if err != nil {
+				errs.append(err)
+			}
+			aggs = append(aggs, agg)
+		}
+	}
+	return aggs, errs.errorOrNil()
+}
+
+// TODO (#3053) Only register callbacks if any instrument matches in a view.
+func (reg *pipelineRegistry[N]) registerCallback(fn func(context.Context)) {
+	for _, pipe := range reg.pipelines {
+		pipe.addCallback(fn)
+	}
+}
+
+type multierror struct {
+	errors []string
+}
+
+func (m *multierror) errorOrNil() error {
+	if len(m.errors) == 0 {
+		return nil
+	}
+	return fmt.Errorf(strings.Join(m.errors, "; "))
+}
+
+func (m *multierror) append(err error) {
+	m.errors = append(m.errors, err.Error())
+}
+
+// instrumentIdentifier is used to identify multiple instruments being mapped to the same aggregator.
+// e.g. using an exact match view with a name=* view.
+// You can't use a view.Instrument here because not all Aggregators are hashable.
+type instrumentIdentifier struct {
+	scope       instrumentation.Scope
+	name        string
+	description string
```

Comment:
How is description part of the identifier? The data model has an explicit statement that description not be treated as identifying. 

https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/metrics/data-model.md#opentelemetry-protocol-data-model

That document considers Unit to be identifying. Should we swap description for unit, here?

Reply:
The description here is used to pass back to the caller any transformed identity.  If you had two views, the default and one that just changed the description, both would be returned here, and it's expected at a higher level to be rejected.  Unit isn't considered here because it's not transformed.

Reply:
If anything we will need to adjust the pipeline's `instrumentKey` and push the description into the value stored in the pipeline's map.

Reply:
https://github.com/open-telemetry/opentelemetry-go/pull/3071 addresses this.